File: packages/core/src/feedback/FeedbackEvent.ts
import { createId } from "../utils/createId";

export type FeedbackEventType =
  | "thumbs_up"
  | "thumbs_down"
  | "action_success"
  | "action_failed"
  | "memory_saved"
  | "privacy_redacted"
  | "session_completed";

export interface FeedbackEvent {
  id: string;
  type: FeedbackEventType;
  sessionId: string;
  data?: Record<string, unknown>;
  timestamp: string;
}

export function createFeedbackEvent(
  type: FeedbackEventType,
  sessionId: string,
  data?: Record<string, unknown>
): FeedbackEvent {
  return {
    id: createId("feedback"),
    type,
    sessionId,
    data,
    timestamp: new Date().toISOString(),
  };
}

File: packages/compose/src/main/kotlin/ai/mobileagentos/compose/feedback/FeedbackEvent.kt
package ai.mobileagentos.compose.feedback

import ai.mobileagentos.compose.utils.IdGenerator

enum class FeedbackEventType {
    USER_INTERACTION,
    PRIVACY_REDACTED,
    ACTION_EXECUTED,
    ERROR_OCCURRED,
    SESSION_COMPLETED,
    CUSTOM
}

data class FeedbackEvent(
    val id: String,
    val type: FeedbackEventType,
    val sessionId: String,
    val timestamp: String,
    val data: Map<String, Any>? = null
) {
    companion object {
        fun create(
            type: FeedbackEventType,
            sessionId: String,
            data: Map<String, Any>? = null
        ): FeedbackEvent {
            return FeedbackEvent(
                id = IdGenerator.generate("feedback"),
                type = type,
                sessionId = sessionId,
                timestamp = System.currentTimeMillis().toString(),
                data = data
            )
        }
    }
}